### ISSUE TITLE
Use begin/end rules for comments and strings

### ISSUE THREAD

**Issue by aeschli** [Microsoft/TypeScript-TmLanguage#181]
TextMate injection grammars make it possible to inject syntax highlighting inside scopes like comments and string.
In order to allow that, grammars must use begin/end rules rather than match, as grammar injections are applied to rules not scopes.

E.g
Instead of
```xml
		<key>comment-line</key>
		<dict>
			<key>match</key>
			<string>(//).*$\n?</string>
			<key>name</key>
			<string>comment.line.ts</string>
		</dict>
```
use
```xml
       <key>comment-line</key>
	<dict>
		<key>begin</key>
		<string>//</string>
		<key>end</key>
		<string>$\n?</string>
		<key>name</key>
		<string>comment.line.ts</string>
	</dict>
```

I made a quick pass on the TypeScript grammar and found that the single line comment needs to be fixed.
Block comment & strings seem to do fine.

**Comment by googol7** [comment]
The same applies for Atom. An issue already exists there: https://github.com/TypeStrong/atom-typescript/issues/848

I already made this PR https://github.com/TypeStrong/atom-typescript/pull/1035 but it would be better to fix the two files:

https://github.com/Microsoft/TypeScript-TmLanguage/blob/master/TypeScript.YAML-tmLanguage

https://github.com/Microsoft/TypeScript-TmLanguage/blob/master/TypeScriptReact.YAML-tmLanguage

**Comment by sheetalkamat** [comment]
Fixed by #257
